glos_qartod/run.py

```diff
@@ -57,41 +57,46 @@ def qc_subset(dir_root, conf, mappings):
 
 
 def find_files(dest_dir, qc_varnames, qc_varnames_bkp):
-    files = []
+    nc_files = []
     if os.path.exists(dest_dir):
         for root, subdir, fname in os.walk(dest_dir):
-            for nc_file in (f for f in fname if f.endswith('.nc')):
-                file_dest = os.path.join(root, nc_file)
-                # check if all the variables already exist in the file
-                # if all of the variables for the expected QC variables
-                # aren't present in the dataset, add the file to the list
-                # of files to be QCed
+            # find all .nc files in this directory, check if there has been
+            # qc applied to them, and create absolute paths to them
+            # path to them
+
+            # higher order function to use in checking listcomp
+            def pred_fn(filepath):
+                return check_if_qc_vars_exist(os.path.join(root, filepath),
+                                              qc_varnames, qc_varnames_bkp)
 
-                qc_filepath = file_dest.rsplit('.', 1)[0] + '.ncq'
-                # try to fetch the QC file's variable names.  If it does not
-                # exist, no QC has been applied and it must be created later
-                if os.path.exists(qc_filepath):
-                    try:
-                        with Dataset(qc_filepath) as f:
-                            qc_vars = f.variables.keys()
-                    # if for some reason we can't open the file,
-                    # note the exception and treat the qc variables as empty
-                    except Exception as e:
-                        get_logger().error('Failed to open file %s: %s',
-                                           qc_filepath, str(e))
-                        qc_vars = []
-                else:
-                    qc_vars = []
-                # check if all the QC variables exist in the file.
-                # if they don't, add them to the list of files to be processed
-                if not (qc_varnames.issubset(qc_vars) or
-                        qc_varnames_bkp.issubset(qc_vars)):
-                    files.append(file_dest)
+            nc_files.extend([os.path.join(root, f) for f in fname
+                             if f.endswith('.nc') and not pred_fn(root)])
```

Comment:
Please replace this list comprehension and dynamically created function with a plain for loop appending to a list of that file is a netCDF file. This code isn't clear, and if it took me this long to figure out what it does even with the comments, it's too complicated. 

```python
nc_files = []
for root, subdir, fname in os.walk(dest_dir):
    if not fname.endswith('.nc'):
        continue

    full_path = os.path.join(root, fname)
    if not check_if_qc_vars_exist(full_path, qc_varnames, qc_varnames_bkup):
        nc_files.append(full_path)

```